### PR TITLE
enh(api): remove install directory after update

### PR DESCRIPTION
**Pull request by kduret**
## Description

remove install directory afetr update

**Fixes** MON-12296

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x
- [x] 22.10.x (master)

### CODE REVIEW

File: tests/php/Core/Platform/Infrastructure/Repository/FsReadUpdateRepositoryTest.php [hunk 1]
```diff
@@ -24,14 +24,33 @@
...
+use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Finder\Finder;
 
 beforeEach(function () {
+    $this->filesystem = $this->createMock(Filesystem::class);
     $this->finder = $this->createMock(Finder::class);
 });
 
+it('should not find updates if install directory does not exist', function () {
+    $repository = new FsReadUpdateRepository($this->filesystem ,$this->finder);
```
Comment (kduret) [inline tests/php/Core/Platform/Infrastructure/Repository/FsReadUpdateRepositoryTest.php]:
```suggestion
    $repository = new FsReadUpdateRepository($this->filesystem, $this->finder);
```

File: tests/php/Core/Platform/Infrastructure/Repository/FsReadUpdateRepositoryTest.php [hunk 2]
```diff
@@ -24,14 +24,33 @@
...
+        ->method('exists')
+        ->willReturn(false);
+
+    $availableUpdates = $repository->findOrderedAvailableUpdates('22.04.0');
+    expect($availableUpdates)->toEqual([]);
+});
+
 it('should order found updates', function () {
-    $repository = new FsReadUpdateRepository($this->finder);
+    $repository = new FsReadUpdateRepository($this->filesystem ,$this->finder);
```
Comment (kduret) [inline tests/php/Core/Platform/Infrastructure/Repository/FsReadUpdateRepositoryTest.php]:
```suggestion
    $repository = new FsReadUpdateRepository($this->filesystem, $this->finder);
```

File: src/Core/Platform/Application/UseCase/UpdateVersions/UpdateVersions.php [hunk 3]
```diff
@@ -166,4 +168,20 @@ private function runUpdates(array $versions): void
             }
         }
     }
+
+    /**
+     * Run post update actions
+     *
+     * @param string $currentVersion
+     */
```
Comment (kduret) [inline src/Core/Platform/Application/UseCase/UpdateVersions/UpdateVersions.php]:
```suggestion
     * @param string $currentVersion
     *
     * @throws UpdateVersionsException
     */
```

File: src/Core/Platform/Infrastructure/Repository/DbWriteUpdateRepository.php [hunk 4]
```diff
@@ -59,6 +65,62 @@ public function runUpdate(string $version): void
...
+        );
+    }
+
+    /**
+     * Remove installation directory
+     */
+    private function removeInstallDirectory(): void
+    {
+        $this->info(
+            "Removing installation directory...",
```
Comment (kduret) [inline src/Core/Platform/Infrastructure/Repository/DbWriteUpdateRepository.php]:
```suggestion
            "Removing installation directory",
```

File: src/Core/Platform/Infrastructure/Repository/DbWriteUpdateRepository.php [hunk 5]
```diff
@@ -59,6 +65,62 @@ public function runUpdate(string $version): void
...
+    {
+        $centreonVarLibPath = $this->parameterBag->get('centreon_var_lib');
+        if (! is_string($centreonVarLibPath)) {
+            throw new \Exception('Cannot get centreon var lib path from configuration');
+        }
+
+        $backupDirectory = $centreonVarLibPath . '/installs/install-' . $currentVersion . '-' . date('Ymd_His');
+
+        $this->info(
+            "Backing up installation directory...",
```
Comment (kduret) [inline src/Core/Platform/Infrastructure/Repository/DbWriteUpdateRepository.php]:
```suggestion
            "Backing up installation directory",
```

File: src/Core/Platform/Infrastructure/Repository/DbWriteUpdateRepository.php [hunk 6]
```diff
@@ -39,10 +41,14 @@ class DbWriteUpdateRepository extends AbstractRepositoryDRB implements WriteUpda
...
      * @param Container $dependencyInjector
      * @param DatabaseConnection $db
+     * @param Filesystem $filesystem
+     * @param ParameterBagInterface $parameterBag
      */
     public function __construct(
         private Container $dependencyInjector,
         DatabaseConnection $db,
+        private Filesystem $filesystem,
+        private ParameterBagInterface $parameterBag,
```
Comment (kduret) [inline src/Core/Platform/Infrastructure/Repository/DbWriteUpdateRepository.php]: inject directly parameter

**Review by jeremyjaouen — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**